File: Models/Coin.py
from pydantic import BaseModel
from typing import Optional

class Coin(BaseModel):
    name: str
    symbol: str
    ipfs_url: str
    telegram_url: Optional[str] = None
    twitter_url: Optional[str] = None
    website_url: Optional[str] = None
    image_url: Optional[str] = None

File: backend/internal/monitoring/models/coin.go
package models

type DecodedCreateInstruction struct {
	Name    string
	Symbol  string
	IpfsUrl string
}

type MintData struct {
	Signature        string
	Name             string
	Symbol           string
	IpfsUrl          string
	TokenAddr        string
	CreatorAddr      string
	BondingCurveAddr string
	DevHoldingAmount float64
}

type IPFS struct {
	ImageURL    string `json:"image"`
	TwitterURL  string `json:"twitter"`
	TelegramURL string `json:"telegram"`
	WebsiteURL  string `json:"website"`
}

// Coin Compositions of the two structs above
type Coin struct {
	CoinData MintData
	IPFSData IPFS
}
